source/extensions/common/wasm/wasm.h

```diff
@@ -141,6 +141,14 @@ class Wasm : public Logger::Loggable<Logger::Id::wasm>, public std::enable_share
     return true;
   }
 
+  void addAfterVmCallAction(std::function<void()> f) { after_vm_call_actions_.push_back(f); }
```

Comment:
You're both adding at and removing from the back of the queue, which makes this LIFO. I think it should be FIFO instead.

Reply:
Done.